hipe4ml/tree_handler.py

```diff
@@ -0,0 +1,278 @@
+        size: int
+            Number of candidates to return. Cannot be used with frac. Default = 1 if
+            frac = None.
+
+        frac: float
+            Fraction of candidates to return. Cannot be used with n.
+
+        inplace: bool
+            If True the shuffled dataframe replaces the initial dataframe. Otherwise return a copy
+            of the shuffled df
+
+        Returns
+        ------------------------------------------------
+        out: pandas.DataFrame or None
+            If inplace == True None is returned and the full DataFrame is replaced
+
+        """
+        df_shuf = None
+        if inplace:
+            self._full_data_frame = self._full_data_frame.sample(size, frac)
+            return df_shuf
+
+        df_shuf = self._full_data_frame.sample(size, frac)
+        return df_shuf
+
+    def eval_data_frame(self, ev_str, inplace=True):
+        """
+        Evaluate a string describing operations on DataFrame columns
+
+        Parameters
+        ------------------------------------------------
+        preselection: str
+            The expression string to evaluate. A combination of a date and a time. Attributes: ()
+            The string syntax is the one required in the pandas.DataFrame.eval() method.
+        inplace: bool
+            If the expression contains an assignment, whether to perform the operation inplace and
+            mutate the existing DataFrame. Otherwise, a new DataFrame is returned.
+
+        Returns
+        ------------------------------------------------
+        out: pandas.DataFrame or None
+            if inplace == True None is returned and the full dataframe is evaluated
+
+        """
+        df_ev = None
+        if inplace:
+            self._full_data_frame.eval(ev_str, inplace=True)
+            return df_ev
+
+        df_ev = self._full_data_frame.eval(ev_str)
+        return df_ev
+
+    def print_summary(self):
+        """
+        Print information about the DataHandler object and its
+        data members
+        """
+        print("\nFile name: ", self._file)
+        print("Tree name: ", self._tree)
+        print("DataFrame head:\n", self._full_data_frame.head(5))
+        print("\nPreselections:", self._preselections)
+        print("Sliced variable: ", self._projection_variable)
+        print("Slices binning: ", self._projection_binning)
+
+    def write_df_to_parquet_files(self, base_file_name="TreeDataFrame", path="./", save_slices=False):
+        """
+        Write the pandas dataframe to parquet files
+
+        Parameters
+        ------------------------------------------------
+        base_file_name: str
+            Base filename used to save the parquet files
+        path: str
+            Base path of the output files
+        save_slices: bool
+            If True and the slices are available, single parquet files for each
+            bins are created
+        """
+        self._full_data_frame.to_parquet(
```

Comment:
What happens here if the original dataframe has been deleted by `slice_data_frame(..)` ?